packages/@react-stately/tooltip/test/useTooltipTriggerState.test.js

```diff
@@ -0,0 +1,200 @@
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  afterEach(() => {
+    onOpenChange.mockClear();
+    // there's global state, so we need to make sure to run out the cooldown for every test
+    act(() => {jest.runAllTimers();});
+  });
+
+  describe('custom close delay', () => {
+    it('closes with a custom close delay less than the default', () => {
+      let closeDelay = 100;
+  
+      let {getByRole, queryByRole, getByLabelText} = render(
+        <TooltipTrigger onOpenChange={onOpenChange} closeDelay={closeDelay} tooltip="Helpful information">
+          Trigger
+        </TooltipTrigger>
+      );
+      fireEvent.mouseDown(document.body);
+      fireEvent.mouseUp(document.body);
+
+      let button = getByLabelText('trigger');
+      fireEvent.mouseEnter(button);
+      fireEvent.mouseMove(button);
+
+      expect(onOpenChange).not.toHaveBeenCalled();
+      expect(queryByRole('tooltip')).toBeNull();
+
+      // run through open timer and confirm that it has opened
+      act(() => jest.advanceTimersByTime(TOOLTIP_DELAY));
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+      let tooltip = getByRole('tooltip');
+      expect(tooltip).toBeVisible();
+
+      fireEvent.mouseLeave(button);
+      fireEvent.mouseMove(button);
+
+      // run the rest of the way through the timer and confirm that it has closed
+      // Note that closeDelay is less than the default close time of 500ms
+      act(() => jest.advanceTimersByTime(closeDelay / 2));
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+    });
+
+    it('closes with a custom close delay more than the default', () => {
+      let closeDelay = 650;
+      let delay = 350;
+
+      let {getByRole, queryByRole, getByLabelText} = render(
+        <TooltipTrigger onOpenChange={onOpenChange} closeDelay={closeDelay} delay={delay} tooltip="Helpful information">
+          Trigger
+        </TooltipTrigger>
+      );
+      fireEvent.mouseDown(document.body);
+      fireEvent.mouseUp(document.body);
+
+      let button = getByLabelText('trigger');
+      fireEvent.mouseEnter(button);
+      fireEvent.mouseMove(button);
+
+      expect(onOpenChange).not.toHaveBeenCalled();
+      expect(queryByRole('tooltip')).toBeNull();
+
+      // run through open timer and confirm that it has opened
+      act(() => jest.advanceTimersByTime(delay));
+      expect(onOpenChange).toHaveBeenCalledWith(true);
+      let tooltip = getByRole('tooltip');
+      expect(tooltip).toBeVisible();
+
+      fireEvent.mouseLeave(button);
+
+      // run the timer to the default close time to make sure it hasn't closed
+      act(() => jest.advanceTimersByTime(TOOLTIP_COOLDOWN));
+      expect(onOpenChange).not.toHaveBeenCalledWith(false);
+      expect(tooltip).toBeVisible();
+
+      // run the rest of the way through the timer and confirm that it has closed
+      // Note that closeDelay is more than the default close time of 500ms
+      act(() => jest.advanceTimersByTime(closeDelay / 2));
```

Comment:
Perhaps this should be `TOOLTIP_COOLDOWN - closeDelay` to advance the rest of the time for the custom cooldown?

Reply:
Yep, good catch